src/main/scala/com/mozilla/telemetry/pings/MainPing.scala

```diff
@@ -107,6 +133,28 @@ case class MainPing(application: Application,
     dynamicProcessEvents.filter(_.category == "normandy")
   }
 
+
+  override def sessionSplitEvents: Seq[JObject] = {
+    import org.json4s.JsonDSL._
+
+    val eventType = "session split"
+    val event = pingAmplitudeProperties merge
```

Comment:
Two more points re: top-level vs "normalish" event
- The top-level field only makes sense if main pings are included in the doctype field, which may or may not be the case for many of these jobs going forward especially as event pings go to release.  What should the behavior be if `includeSessionSplitEvent` is true but the `docType` filter doesn't include `main`? 
- There's already a bit of a divergence in logic here (sessionStart calculations here vs in `eventToAmplitudeEvent`). I can only imagine this getting worse over time. If you'd like to keep the top-level field, I think it'd be a good idea to add a comment in `eventToAmplitudeEvent` about checking over this method to make sure the changes shouldn't be reflected here. I still think that's a sign this should be reworked -- if you don't want to go with specifying an event in the schema then it would still make sense to create `Event` and `AmplitudeEvent` objects and pass them through the common formatter.

Reply:
>  What should the behavior be if includeSessionSplitEvent is true but the docType filter doesn't include main

The sessionSplitEvent is empty by default and has to be overridden specifically by different ping types. So it's well defined that behavior in this case is that no sessionSplitEvent is produced.

We'd have the same problem if we injected a special event for the session split and let it flow through. That special event needs access to fields specific to the ping type, so we'd need to define the event per ping type, likely making the default to be no event. So there would be the possibility of configuring the schema to pick up the event, but the Scala code doesn't implement any event to match, so would have no effect.

> it would still make sense to create Event and AmplitudeEvent objects and pass them through the common formatter.

I'm playing with this idea.

Reply:
Okay, one more thought/piece of context that might not be clear: there is no such thing as a session split in mobile. This is a concept that's specific to the main ping, hence why it feels wrong to me to add this at the top-level to a schema that includes non-desktop pings.

> So it's well defined that behavior in this case is that no sessionSplitEvent is produced.

My point is that if that happens it's likely an error by the person writing the schema. If we want to get to a point where it's not data engineers writing the schema (which, dear deity I hope we do) we should consider where we might be putting footguns ;)